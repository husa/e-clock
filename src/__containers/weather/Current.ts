import { connect } from 'react-redux';

import { selectLocation, selectCurrentWeather, selectSettings } from '../../__selectors';

import CurrentWeather from '../../components/weather/Current';

const mapStateToProps = (state) => ({
  data: selectCurrentWeather(state),
  location: selectLocation(state),
  temperatureUnits: selectSettings(state).temperatureUnits,
});

export default connect(mapStateToProps)(CurrentWeather);
