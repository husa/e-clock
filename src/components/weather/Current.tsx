import './current.scss';

import Icon from './Icon';
import type { WeatherResponseCurrent, WeatherResponseLocation } from '../../types';

type Props = {
  data: WeatherResponseCurrent;
  location: WeatherResponseLocation;
  temperatureUnits: 'c' | 'f';
};

const CurrentWeather = ({ data, location, temperatureUnits }: Props) => (
  <div className="weather-current">
    <div className="weather-current__city">{location.city}</div>
    <Icon className="weather-current__icon" code={data.code} title={data.text} />
    <div className="weather-current__temperature">{data.temp}</div>
    <div className="weather-current__degrees">Â°{temperatureUnits.toUpperCase()}</div>
  </div>
);

export default CurrentWeather;
