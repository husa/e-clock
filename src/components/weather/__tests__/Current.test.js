import React from 'react';
import renderer from 'react-test-renderer';

import Current from '../Current';


describe('Components > weather > Current', () => {
  it('should match snapshot', () => {
    const testData = {
      code: '1',
      date: '18 Jan 2018',
      temp: -273,
      text: 'test-text'
    };

    const testLocation = {
      city: 'TestCity'
    };

    const testTemperatureUnits = 'test-temp-unit';


    const component = renderer.create(
      <Current
        data={testData}
        location={testLocation}
        temperatureUnits={testTemperatureUnits} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
