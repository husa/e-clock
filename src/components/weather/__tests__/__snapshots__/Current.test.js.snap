// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > weather > Current should match snapshot 1`] = `
<div
  className="weather-current"
>
  <div
    className="weather-current__city"
  >
    TestCity
  </div>
  <img
    alt="test-text"
    className="weather-icon weather-current__icon"
    src="/img/weather-icons/weather_1.png"
    title="test-text"
  />
  <div
    className="weather-current__temperature"
  >
    -273
  </div>
  <div
    className="weather-current__degrees"
  >
    Â°
    TEST-TEMP-UNIT
  </div>
</div>
`;
